controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -417,24 +428,35 @@ private ReaderGroupConfiguration getRGConfigurationFromRecord(final String scope
-            return CompletableFuture.completedFuture(CreateReaderGroupStatus.Status.INVALID_RG_NAME);
+            return CompletableFuture.completedFuture(CreateReaderGroupResponse.newBuilder()
+                    .setStatus(CreateReaderGroupResponse.Status.INVALID_RG_NAME).build());
         }
         return RetryHelper.withRetriesAsync(() -> {
             // 1. check if scope with this name exists...
             return streamMetadataStore.checkScopeExists(scope)
                     .thenCompose(exists -> {
                         if (!exists) {
-                            return CompletableFuture.completedFuture(CreateReaderGroupStatus.Status.SCOPE_NOT_FOUND);
+                            return CompletableFuture.completedFuture(CreateReaderGroupResponse.newBuilder()
+                                    .setStatus(CreateReaderGroupResponse.Status.SCOPE_NOT_FOUND).build());
                         }
                         //2. check state of the ReaderGroup
                         return isRGCreationComplete(scope, rgName)
                                 .thenCompose(complete -> {
                                     if (!complete) {
                                         return streamMetadataStore.addReaderGroupToScope(scope, rgName, config.getReaderGroupId())
                                                 .thenCompose(x -> createReaderGroupTasks(scope, rgName, config, createTimestamp))
-                                                .thenApply(y -> CreateReaderGroupStatus.Status.SUCCESS);
+                                                .thenCompose(status -> streamMetadataStore.getReaderGroupId(scope, rgName)
+                                                .thenCompose(rgId -> streamMetadataStore.getReaderGroupConfigRecord(scope, rgName, null, executor)
+                                                        .thenApply(cfgRecord ->
+                                                        CreateReaderGroupResponse.newBuilder()
+                                                                .setConfig(encodeReaderGroupConfigRecord(scope, rgName, cfgRecord.getObject(), rgId))
+                                                                .setStatus(CreateReaderGroupResponse.Status.SUCCESS).build())));
                                     }
-                                    return CompletableFuture.completedFuture(CreateReaderGroupStatus.Status.SUCCESS);
+                                    return streamMetadataStore.getReaderGroupId(scope, rgName)
+                                            .thenCompose(rgId -> streamMetadataStore.getReaderGroupConfigRecord(scope, rgName, null, executor)
+                                                    .thenApply(cfgRecord -> CreateReaderGroupResponse.newBuilder()
+                                                            .setConfig(encodeReaderGroupConfigRecord(scope, rgName, cfgRecord.getObject(), rgId))
```

Comment:
we are calling this encode method twice. perhaps we can compute once and reuse.

Reply:
True.